### PR TITLE
Sinatra prototype running with a sampled JSON

### PR DESCRIPTION
**Pull request by alxngyn**
Info: 
https://github.com/alxngyn/iam/tree/sinatra-setup/sinatra

### CODE REVIEW

File: sinatra/app.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+require 'date'
+
+# preprocess the json from server
+# TODO: change data source to redis
+file = File.read('sample.json')
+data = JSON.parse(file)
+
+# default helloworld
+get '/' do
+  "Hello World!"
```
Comment (ElijahCaine) [inline sinatra/app.rb]: This was put in as a placeholder / cheeky joke. We might want to replace it with a *real* main page; maybe a rendered version of our README or at the very least a simple webpage that points to our project (the osuosl/iam github repo).

File: sinatra/app.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+end
+
+# returns bulk data
+get '/data' do
+  content_type:json
+  data.to_json
```
Comment (ElijahCaine) [inline sinatra/app.rb]: I understand ruby allows us to omit the 'return' statement in favor of just returning the last thing in a block, but I'm in favor of *explicitly* returning data as it is more clear to the reader.

tldr, let's add return statements.

**Comment by ElijahCaine** [comment]
@alxngyn I may have jumped the gun in reviewing this PR, but I wanted to give you some feedback:

## Negatives

1. Fist and foremost I'd like to address the git-flow associated with this PR. I understand that it is the norm for many organizations to fork the project, work on your own fork, and then make a PR from a remote fork into the main repo.

  **This is not the in-house osuosl development workflow**.

  We do all work in the main repository. Neither workflow is more or less correct, it's just a matter of taste and what works for an organization and we currently branch -> merge, not fork -> branch -> merge.

  To clarify, community contributions are welcome to (and actually *must*) fork before making a pull request, but if you are an in-house developer we prefer you work on the main repository. This may change in the future, but it hasn't yet.

2. Your email is set to `pvector@green.workstation.osuosl.bak` in this commit. This means when you login to an osl workstation you need to manually set your email to whatever you have registered with Github the next chance you get. You can retro-actively change the email of a commit after you have set the config option. There are guides for this.

3. This code should not live in a directory called `sinatra/`. Common practice, at least with other frameworks I have used, is to store project code in a directory of the project name. For instance we would rename the `sinatra/` directory to `iam/` or `invoicing-and-metrics/`.

## Positives

1. Your ruby code looks pretty solid. Technically this PR is good.

2. I appreciate that you included a README, documentation is always helpful and yours was specifically useful in contextualizing what I was looking at.

## Lastly

If it feels like I'm being mean / picking on you I'd like to apologize. This is all necessary feedback and it will make future PR's go a lot smoother.

I have also included some inline comments related to the content of this PR. Good work on making a Pull Request and lets hope it is the first of many!

File: sinatra/app.rb (hunk 2 above)
Comment (matthewrsj) [inline sinatra/app.rb]: I agree with @ElijahCaine here. Leaving out the explicit return is considered by many to be the `ruby way`, but it can be confusing and problematic. See: http://stackoverflow.com/a/13635060

**Comment by alxngyn** [comment]
Thanks for the feedback @ElijahCaine 

### Negatives:
* OK I didn't know there was an official OSL workflow, I'll create a branch in this repo. Also is there a doc on this? I didn't see anything on the Dev Team section on the docs.
* Forgot to do  git.config, Fixed!
* Since the architecture of the project isn't defined yet, I thought labeling the folder name sinatra would help segment everything until we have a clear idea of the main app. I'll fix this when I push to the main repo.

There's no need to feel sorry about feedback. I'm new and I was expecting it on my first few PRs.

**Comment by matthewrsj** [comment]
@alxngyn 
This entire doc is a great resource, and the linked section answers your specific question: http://wiki.osuosl.org/development_process/1_development.html#the-development-process

**Comment by alxngyn** [comment]
Whoa I didn't know there was a wiki. I was told to go over the docs @ docs.osuosl.org
Thanks @matthewrsj!

**Comment by alxngyn** [comment]
Closing this PR since we already have the other sinatra framework running
